src/executor/cursor.rs

```diff
@@ -363,6 +408,231 @@ mod tests {
         Ok(())
     }
 
+    #[smol_potat::test]
+    async fn first_2_before_10_also_related_select() -> Result<(), DbErr> {
```

Comment:
These tests are cool! Would appreciate a few integration tests (in sqlite/mysql/postgres)